File: src/main/java/dev/lampirg/consultationappointment/web/fetch/ConsultationPattern.java
package dev.lampirg.consultationappointment.web.fetch;

import dev.lampirg.consultationappointment.data.teacher.Teacher;
import jakarta.persistence.Embeddable;
import jakarta.persistence.Embedded;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;

@Data
@Embeddable
public class ConsultationPattern {

    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    private LocalDate until;
    private LocalDate fromDate;
    @Embedded
    private ConsultationInfo consultationInfo;
    @ManyToOne
    @JoinColumn(name = "teacher_id")
    private Teacher teacher;

}

File: src/main/kotlin/dev/lampirg/consultationappointment/web/fetch/ConsultationPattern.kt
package dev.lampirg.consultationappointment.web.fetch

import dev.lampirg.consultationappointment.data.teacher.Teacher
import jakarta.persistence.Embeddable
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import lombok.Data
import org.springframework.format.annotation.DateTimeFormat
import java.time.LocalDate

@Data
@Embeddable
class ConsultationPattern(
    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE)
    var until: LocalDate,
    var fromDate: LocalDate,
    var consultationInfo: ConsultationInfo,
    @ManyToOne
    @JoinColumn(name = "teacher_id")
    var teacher: Teacher? = null
) {
    constructor() : this(LocalDate.MIN, LocalDate.MIN, ConsultationInfo())
}